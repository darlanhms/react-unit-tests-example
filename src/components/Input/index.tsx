import styles from "./styles.module.css";

interface InputProps extends React.HTMLAttributes<HTMLInputElement> {
  label?: string;
}

const Input: React.FC<InputProps> = ({ label, ...rest }) => {
  return (
    <div className={styles.container}>
      {label && <label className={styles.label}>{label}</label>}
      <input {...rest} />
    </div>
  );
};

export default Input;
