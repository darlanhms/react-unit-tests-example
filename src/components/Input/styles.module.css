.container {
  display: flex;
  flex-direction: column;
}

.label {
  font-weight: bold;
}
