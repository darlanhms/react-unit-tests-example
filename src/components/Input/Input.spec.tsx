import { render, screen } from "@testing-library/react";
import Input from ".";

describe("Input", () => {
  it("matches snapshot", () => {
    const { container } = render(<Input />);

    const input = screen.getByRole("textbox");

    expect(input).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("renders a label when informed", () => {
    const { container } = render(<Input label="Nome" />);

    const input = screen.getByRole("textbox");
    const label = screen.getByText("Nome");

    expect(input).toBeInTheDocument();
    expect(label).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
