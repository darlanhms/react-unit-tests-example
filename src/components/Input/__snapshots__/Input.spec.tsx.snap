// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches snapshot 1`] = `
<div>
  <div>
    <input />
  </div>
</div>
`;

exports[`Input renders a label when informed 1`] = `
<div>
  <div>
    <label>
      Nome
    </label>
    <input />
  </div>
</div>
`;
